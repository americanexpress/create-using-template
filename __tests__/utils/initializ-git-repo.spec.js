/*
 * Copyright 2021 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations
 * under the License.
 */

const initializeGitRepo = require('../../src/utils/initialize-git-repo');
const runCommand = require('../../src/utils/run-command');

jest.mock('../../src/utils/run-command', () => jest.fn());

describe('initializeGitRepo', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.spyOn(console, 'log').mockImplementation(() => {});
  });
  it('should initialize a git repo with a passable commit message on the main branch', async () => {
    await initializeGitRepo('repoPathMock', {});

    expect(runCommand.mock.calls).toMatchSnapshot();
  });

  it('should initialize a git repo with a message and a branch from the special template values', async () => {
    await initializeGitRepo('repoPathMock', {
      _defaultBranchName: 'primaryBranchNameMock',
      _initialCommitMessage: 'initialCommitMessageMock',
    });

    expect(runCommand.mock.calls).toMatchSnapshot();
  });
});
