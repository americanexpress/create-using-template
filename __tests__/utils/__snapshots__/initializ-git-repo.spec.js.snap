// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initializeGitRepo should initialize a git repo with a message and a branch from the special template values 1`] = `
Array [
  Array [
    "git",
    Array [
      "init",
    ],
    "repoPathMock",
  ],
  Array [
    "git",
    Array [
      "add",
      ".",
    ],
    "repoPathMock",
  ],
  Array [
    "git",
    Array [
      "commit",
      "-minitialCommitMessageMock",
      "--quiet",
    ],
    "repoPathMock",
  ],
  Array [
    "git",
    Array [
      "branch",
      "-m",
      "primaryBranchNameMock",
    ],
    "repoPathMock",
  ],
]
`;

exports[`initializeGitRepo should initialize a git repo with a passable commit message on the main branch 1`] = `
Array [
  Array [
    "git",
    Array [
      "init",
    ],
    "repoPathMock",
  ],
  Array [
    "git",
    Array [
      "add",
      ".",
    ],
    "repoPathMock",
  ],
  Array [
    "git",
    Array [
      "commit",
      "-mfeat(generation): initial commit",
      "--quiet",
    ],
    "repoPathMock",
  ],
  Array [
    "git",
    Array [
      "branch",
      "-m",
      "main",
    ],
    "repoPathMock",
  ],
]
`;
